ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -376,14 +402,48 @@ - (NSDictionary *)constantsToExport
     callback(@[[NSNumber numberWithBool:isPinOrFingerprintSet]]);
 }
 
+- (void)batteryLevelDidChange:(NSNotification *)notification
+{
+    float batteryLevel = [self.powerState[@"batteryLevel"] floatValue];
+    [self sendEventWithName:@"batteryLevelDidChange" body:@(batteryLevel)];
+
+    if (batteryLevel <= _lowBatteryThreshold) {
+        [self sendEventWithName:@"batteryLevelIsLow" body:@(batteryLevel)];
+    }
+}
+
+- (void)powerStateDidChange:(NSNotification *)notification
+{
+    [self sendEventWithName:@"powerStateDidChange" body:self.powerState];
+}
+
+- (NSDictionary *)powerState
+{
+    return @{
+#if TARGET_OS_TV
+             @"batteryLevel": @1,
+             @"batteryState": @"full",
+#else
+             @"batteryLevel": @([UIDevice currentDevice].batteryLevel),
+             @"batteryState": [@[@"unknown", @"unplugged", @"charging", @"full"] objectAtIndex: [UIDevice currentDevice].batteryState],
```

Comment:
Just added a warning log if running in dev mode and on an actual device (IIRC from testing the simulator always returns `unknown` so this could be confusing if you've already enabled the battery monitoring mode)

Reply:
I see the warning, that is helpful. I do much of my testing on an iOS emulator so I'll notice, if you can confirm the emulator always returns unknown since it is a dev notice anyway maybe you could add that?  "Emulators typically return unknown every time." or whatever. The pain now will avoid 20 duplicate issues over 2 years I bet

Reply:
The emulator does return unknown in my testing, if you accept my PR to your fork (which will then push my suggested warning changes into this PR) then this will be resolved

Reply:
thanks, resolved this one